labrad/server.py

```diff
@@ -350,6 +350,7 @@ def startup(self, protocol):
             log.err("connection failed, disconnecting")
             traceback.print_exc()
             self.disconnect(e)
+            raise
```

Comment:
Shouldn't you have some specific type of error here (perhaps `RuntimeError`) which is at least slightly descriptive and so the user can specifically choose to swallow the error if needed? Just having `raise` gives the following, which doesn't seem relevant to the situation at hand:
```
In [1]: raise
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-26814ed17a01> in <module>()
----> 1 raise

TypeError: exceptions must be old-style classes or derived from BaseException, not NoneType
```

Further, do you ever actually print out the full error traceback, since that can be useful (maybe done in `traceback.print_exc()`)?

Finally, I don't know if doing this at all is a good idea; I'll let someone more experienced (e.g., Dan, Evan, Julian) decide.